utils/streamlit_utils.py

```diff
@@ -294,78 +432,85 @@ def expander_npmi_description(min_vocab):
-            )
 
-            alpha_warning = "Your alpha value is a bit on the high side, which means that the distribution over words in this dataset is a bit unnatural. This could be due to non-language items throughout the dataset."
-            xmin_warning = "The minimum rank for this fit is a bit on the high side, which means that the frequencies of your most common words aren't distributed as would be expected by Zipf's law."
-            fit_results_table = pd.DataFrame.from_dict(
-                {
-                    r"Alpha:": [str("%.2f" % z.alpha)],
-                    "KS distance:": [str("%.2f" % z.ks_distance)],
-                    "Min rank:": [str("%s" % int(z.xmin))],
-                },
-                columns=["Results"],
-                orient="index",
-            )
-            fit_results_table.index.name = column_id
-            st.caption(
-                "Use this widget for the counts of different words in your dataset, measuring the difference between the observed count and the expected count under Zipf's law."
-            )
-            st.markdown(_ZIPF_CAPTION)
-            st.write(
-                """
-            A Zipfian distribution follows the power law: $p(x) \propto x^{-α}$
-    with an ideal α value of 1."""
-            )
-            st.markdown(
-                "In general, an alpha greater than 2 or a minimum rank greater than 10 (take with a grain of salt) means that your distribution is relativaly _unnatural_ for natural language. This can be a sign of mixed artefacts in the dataset, such as HTML markup."
-            )
-            st.markdown(
-                "Below, you can see the counts of each word in your dataset vs. the expected number of counts following a Zipfian distribution."
-            )
-            st.markdown("-----")
-            st.write("### Here is your dataset's Zipf results:")
-            st.dataframe(fit_results_table)
-            st.write(zipf_summary)
-            # TODO: Nice UI version of the content in the comments.
-            # st.markdown("\nThe KS test p-value is < %.2f" % z.ks_test.pvalue)
-            # if z.ks_test.pvalue < 0.01:
-            #    st.markdown(
-            #        "\n Great news! Your data fits a powerlaw with a minimum KS " "distance of %.4f" % z.distance)
-            # else:
-            #    st.markdown("\n Sadly, your data does not fit a powerlaw. =(")
-            # st.markdown("Checking the goodness of fit of our observed distribution")
-            # st.markdown("to the hypothesized power law distribution")
-            # st.markdown("using a Kolmogorov–Smirnov (KS) test.")
-            st.plotly_chart(zipf_fig, use_container_width=True)
-            if z.alpha > 2:
-                st.markdown(alpha_warning)
-            if z.xmin > 5:
-                st.markdown(xmin_warning)
-        except:
-            st.write("Under construction!")
+    #with st.expander(
```

Comment:
It looks like what you're doing here is removing the `with st.expander` logic, and, in app.py, adding `with tab` logic. Would it be straightforward to have the `with tab` logic here, replacing the `with st.expander` logic? If so, I would recommend doing that instead, since these `with` statements are commands for the streamlit UI -- and so should ideally be in the streamlit_utils.py file, if possible.

Reply:
ohhh okay, now I get what you mean 👍